src/app/locationForm/location-name/LocationName.jsx

```diff
@@ -12,6 +13,14 @@ class LocationName extends Component {
     this.onConfirm = this.onConfirm.bind(this);
     this.onEdit = this.onEdit.bind(this);
     this.onSubmit = this.onSubmit.bind(this);
+
+    if (!props.locationData) {
+      props.getLocation(props.match.params.locationId);
+    }
+  }
+
+  componentWillReceiveProps(props){
```

Comment:
I'm not sure this is the behavior we want. Once the screen has rendered and the user may be interacting with it, suddenly changing modes automatically as data comes in may be weird, and also could create edge cases (e.g. in case the user has started typing something, it might just be wiped).

If we no longer want to assume the data was already fetched in a previous screen (which it currently always should be, except in development with live reload), then we should probably be blocking the UI with some indicator while it loads, rather than letting the user get into confusing situations.

Reply:
Should be fixed in 4565fa5442acafdc7ac23a46bf862b563dd7391e

Reply:
Here's what it looks like:

<img width="415" alt="screen shot 2018-04-21 at 9 32 55 pm" src="https://user-images.githubusercontent.com/586962/39091618-35063d9c-45ac-11e8-82cc-52bd3dd37916.png">
<img width="425" alt="screen shot 2018-04-21 at 9 27 22 pm" src="https://user-images.githubusercontent.com/586962/39091619-39805934-45ac-11e8-99da-c68f51ecee00.png">